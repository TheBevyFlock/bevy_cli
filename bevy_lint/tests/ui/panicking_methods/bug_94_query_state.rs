//! This tests the `panicking_query_methods` lint, specifically when triggered on the `QueryState`
//! type.

#![feature(register_tool)]
#![register_tool(bevy)]
#![deny(bevy::panicking_query_methods)]

use bevy::prelude::*;

#[derive(Component)]
struct Foo;

fn main() {
    let mut world = World::new();

    let mut query_state = QueryState::<&mut Foo>::new(&mut world);

    let _ = QueryState::single(&mut query_state, &world);
    //~^ ERROR:  called a `QueryState` method that can panic when a non-panicking alternative exists
    //~| HELP: use `QueryState::get_single(&mut query_state, &world)`

    QueryState::single_mut(&mut query_state, &mut world);
    //~^ ERROR:  called a `QueryState` method that can panic when a non-panicking alternative exists
    //~| HELP: use `QueryState::get_single_mut(&mut query_state, &mut world)`
}
