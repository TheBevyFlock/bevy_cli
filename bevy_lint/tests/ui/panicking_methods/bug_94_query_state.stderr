error: called a `QueryState` method that can panic when a non-panicking alternative exists
  --> tests/ui/panicking_methods/bug_94_query_state.rs:18:13
   |
18 |     let _ = QueryState::single(&mut query_state, &world);
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `QueryState::get_single(&mut query_state, &world)` and handle the `Option` or `Result`
note: the lint level is defined here
  --> tests/ui/panicking_methods/bug_94_query_state.rs:6:9
   |
6  | #![deny(bevy::panicking_query_methods)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: called a `QueryState` method that can panic when a non-panicking alternative exists
  --> tests/ui/panicking_methods/bug_94_query_state.rs:22:5
   |
22 |     QueryState::single_mut(&mut query_state, &mut world);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: use `QueryState::get_single_mut(&mut query_state, &mut world)` and handle the `Option` or `Result`

error: aborting due to 2 previous errors

